src/Gui/NaviCube.cpp

```diff
@@ -172,13 +172,16 @@ class Face {
 	}
 };
 
-class NaviCubeImplementation {
+class NaviCubeImplementation : public ParameterGrp::ObserverType {
 public:
 	NaviCubeImplementation(Gui::View3DInventorViewer*);
 	virtual ~ NaviCubeImplementation();
 	void drawNaviCube();
 	void createContextMenu(const std::vector<std::string>& cmd);
 
+	/// Observer message from the ParameterGrp
+    virtual void OnChange(ParameterGrp::SubjectType &rCaller,ParameterGrp::MessageType Reason);
```

Comment:
I hate to suggest this (because I am an all-spaces-all-the-time coder), but... the code on either side uses tabs. It would probably be best to stick to that here.